src/Models/TaxonomyTerm.php

```diff
@@ -896,4 +938,114 @@ public function inferReciprocalComponent($remoteClass, $remoteRelation)
 
         return parent::inferReciprocalComponent($remoteClass, $remoteRelation);
     }
+
+
+    /**
+     * @throws \ReflectionException
+     * @return ArrayList
+     */
+    protected function getTaggedDataObjects()
+    {
+        $termID = $this->ID;
+
+        $list = ArrayList::create();
+
+        $classes = ClassInfo::subclassesFor(DataObject::class);
+
+        $relations = ['has_one', 'has_many', 'many_many'];
+
+        foreach ($classes as $class) {
+
+            // exclude self as terms are assigned to terms via hierarchy
+            if ($class === self::class) {
+                continue;
+            }
+
+            foreach ($relations as $relation) {
+                $relationCandidates = Config::inst()->get($class, $relation);
+                if (count($relationCandidates)) {
+                    foreach ($relationCandidates as $field => $fieldType) {
+                        if ($fieldType === self::class) {
+                            $filterField = $field . ($relation === 'has_one' ? '' : '.') . 'ID';
+                            $items       = DataObject::get($class)->filter($filterField, $termID);
+
+                            $items->each(function ($item) use ($list, $field, $relation) {
+                                if (get_class($item) == DataObjectTaxonomyTerm::class) {
+                                    $owner = $item->OwnerObject();
+                                    $fieldName = 'Tags';
```

Comment:
this branch of code is to get the tagged objects defined by `AT` module itself through the polymorphic many_many_through. When we are in this branch of if conditions, we are directly checking DataObjectTaxonomyTerm bridging object,  and the information about what the relation is called (to make $fieldName) can be retrieved from neither the $field nor $relation, nor any available variables at this point, since this piece:
```
class DataObjectTaxonomyTerm extends DataObject {
    private static $has_one = [
        'OwnerObject' => DataObject::class, // Polymorphic has_one
        'JointObject' => TaxonomyTerm::class,
    ];
}
```
and 
```
class DataObjectTaxonomiesDataExtension {
    private static $many_many = [
        'Tags' => [
            'through' => DataObjectTaxonomyTerm::class,
            'from'    => 'OwnerObject',
            'to'      => 'JointObject',
        ],
    ];
}
```
Of cause, this had-coded piece of implementation is based on the assumption that our brideging data object is only used for the the DataObject <=> TaxonomyTerm  relation for AT module, and if Devs added their own many_many_through relation to TaxonomyTerm, they should created their onw bridging data object. 
The current implementation doesn't cover to get generic other $many_many relations if using many_many_through mechanism.

Reply:
let also talk this internally. @michalkleiner

Reply:
I've refactored the function getTaggedDataObjects() so there is no longer hard-coded anything in the latest commit.

Reply:
The function get_cms_link is removed from this module, as it is up to the custom project to solve a CMSEditLink per DataObject.